docusaurus/docs/get-started/anatomy-of-a-plugin.md

```diff
@@ -0,0 +1,181 @@
+
+### Call resource
+
+Apps can have backends to handle server-side functionality, such as making external API calls or processing more advanced authentication methods. The `CallResourceHandler` is commonly used for this purpose. See our [App with backend example](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-with-backend) for implementation details.
+
+### Nested plugins
+
+App plugins can bundle multiple plugins, such as data sources or panels, into a single installable package. This approach is useful for services that require a combination of plugins for full functionality. You can [learn more about working with nested plugins](/how-to-guides/app-plugins/work-with-nested-plugins) in our documentation.
+
+## Data source plugins
+
+Data source plugins allow Grafana to connect to external services, configure queries, and display data. They can include frontend-only or full-stack components (with a backend).
+
+![The components of a Data Source plugin](./images/data-source-plugin.png)
+
+### Config editor
+
+The config editor is where users provide connection details (e.g., API keys, URLs) for the external service. To define the config editor, use `setConfigEditor()` and pass a custom configuration component. You can see [how to define a config editor in our datasource example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-basic/src/components/ConfigEditor/ConfigEditor.tsx).
+
+Ensure sensitive data is stored securely using `secureJson`. Read our guide on [adding authentication for data source plugins](/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins#store-configuration-in-securejsondata) for more details.
+
+### Query editor
+
+The query editor allows users to construct queries against the connected service. This editor is used in the dashboard, Explore mode, and Alerting. Query editors can be customized to provide either a code editor or a guided query builder. You can see [how to define a query editor in our datasource example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-basic/src/components/QueryEditor/QueryEditor.tsx).
+
+### Health check
+
+The "Save and Test" button in the data source config page allows users to verify that the connection works. Plugins can customize this behavior by [adding custom health checks](/how-to-guides/data-source-plugins/convert-a-frontend-datasource-to-backend#health-check).
+
+### Query data
+
+The Query Data method processes multiple queries and returns corresponding responses. Each query includes a `RefID`, which is mapped to its response in a `QueryDataResponse`. The method loops through the queries, processes them individually, and returns either the result or an error with an appropriate status code.
```

Comment:
@andresmgot tagging just in case - would appreciate your review of this section, in particular was wondering about the parallel execution work and whether should be mentioned